docs: Add how-to on shared-deps (#131)
rancher-sandbox/hypper: docs/user/howto/shared-deps.md

# Working with shared dependencies

> Hypper charts understand the concept of shared dependency charts.

Reviewer: ```suggestion
Hypper charts provide a concept of shared dependency charts.
```

Author: :thinking:  mmh, I would say Hypper (or "the implementation") provides the concept of shared dependencies, but the charts "support" them?

Reviewer: support sounds nice...

Author: fixed in https://github.com/rancher-sandbox/hypper/pull/131/commits/6bb8f6dddf819eab4ad0e589bd3285bf59853db0.

A chart declared as a shared dependency is a chart that more than one chart may

> depend on. Charts deployed as shared dependencies are the analogous of system

Reviewer: ```suggestion
depend on, once deployed to a cluster, the deployment may be reused by multiple other deployments. Charts deployed as shared dependencies are the analogous of system
```

Author: fixed in https://github.com/rancher-sandbox/hypper/pull/131/commits/7d9f6332ca1011b74596d6be572a7a0c5c84da5d.

libraries in an OS: dependencies that are used by several consumers. One
deployment may satisfy the shared dependencies of several charts.

## Creating a Hypper chart with shared dependencies

Let's create the simplest chart possible, and add charts as shared dependencies
to it.

First, we create a simple empty chart:

```terminal

> $ helm create our-app

Reviewer: (This is not a blocker) We should port that function over to hypper.

Author: Sounds like a good idea. But even with that, I would like to keep the how-to so people create a hypper chart from a helm chart (instead of calling a command), to know the differences better.

Reviewer: You are right, we should add two sub-chapters later on
- creating a chart with shared dependencies from a helm chart
- creating a hypper chart2

```

Now, we can edit `./our-app/Chart.yaml`, and add some shared dependencies to it:

```diff
apiVersion: v2
name: our-app
description: A Helm chart for Kubernetes
type: application
version: 0.1.0
appVersion: 1.16.0
annotations:
+ hypper.cattle.io/shared-dependencies: |
+   - name: postgresql
+     version: "~8.9.4"
+     repository: "https://charts.bitnami.com/bitnami"
+   - name: prometheus
+     version: "~13.7.0"
+     repository: "https://prometheus-community.github.io/helm-charts"
```

Shared dependencies are just normal Helm
[Dependencies](https://helm.sh/docs/topics/charts/#chart-dependencies). As
such, they are defined with:
- The name field is the name of the chart you want.
- The version field is the version of the chart you want.
- The repository field is the full URL to the chart repository. Note that you

>   must also use helm repo add to add that repo locally. You might use the name

Reviewer 2: hypper instead of helm, otherwise the tutorial wont work :p

Author: woooops! fixed in https://github.com/rancher-sandbox/hypper/pull/131/commits/50f8d779c7984678c6ab6e82b4e60a176a26c51c.
